lib/utils/note-utils.ts

```diff
@@ -42,9 +44,37 @@ export const getTitle = (content) => {
  *
  * @param content
  */
-const getPreview = (content: string) => {
+const getPreview = (content: string, searchQuery?: string) => {
   let preview = '';
   let lines = 0;
+
+  // contextual note previews
+  if (searchQuery?.trim()) {
+    const terms = getTerms(searchQuery);
+
+    // use only the first term of a multi-term query
+    const firstTerm = terms[0].toLocaleLowerCase();
+    const leadingChars = 30 - firstTerm.length;
+
+    const regExp = new RegExp(
+      /* eslint-disable */
+      '(?<=\\s|^)[^\n]' + // split at a word boundary (pattern must be preceded by whitespace or beginning of string)
```

Comment:
we might want to verify that this doesn't exclude all searches from languages that don't use spaces

won't it also eliminate building excerpts for partial-word searches?

Reply:
No, the search still matches anywhere in the note, this just trims the preview to a word boundary. e.g.

<img width="328" alt="Screen Shot 2020-12-11 at 10 00 02 AM" src="https://user-images.githubusercontent.com/52152/101938108-ac322500-3b97-11eb-84e8-03d70c13ab74.png">

Reply:
Here's a kanji search:

<img width="335" alt="Screen Shot 2020-12-11 at 10 00 50 AM" src="https://user-images.githubusercontent.com/52152/101938193-ca982080-3b97-11eb-9d4b-184ffe76bc01.png">